build, microbench-ci: update token retrieval

herkolategan opened the pull request:
This change updates the token retrieval to use gcloud instead of relying on
target_pull_request to provide a token with the required permissions, to add
labels or post to a PR.

Epic: None
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/148194)
<!-- Reviewable:end -->

herkolategan committed:
build, microbench-ci: update token retrieval

This change updates the token retrieval to use gcloud instead of relying on
target_pull_request to provide a token with the required permissions, to add
labels or post to a PR.

Epic: None
Release note: None

cockroach-teamcity commented:
<details><summary><strong>🔴 Sysbench</strong> [SQL, 3node, oltp_read_write]</summary>

| Metric                      | Old Commit     | New Commit     | Delta      | Note         |
|-----------------------------|----------------|----------------|------------|--------------|
| 🔴 **sec/op** | 8.416m ±0% | 8.696m ±0% | +3.33% | p=0.000 n=15    |
| 🔴 **allocs/op** | 9.443k ±0% | 10.10k ±1% | +6.90% | p=0.000 n=15    |

<details><summary>Reproduce</summary>

**benchdiff binaries**:
```shell
mkdir -p benchdiff/c92abdc/bin/1058449141
gcloud storage cp gs://cockroach-microbench-ci/builds/c92abdc71850f94fc69cc6e6453605c3896dc280/bin/pkg_sql_tests benchdiff/c92abdc/bin/1058449141/cockroachdb_cockroach_pkg_sql_tests
chmod +x benchdiff/c92abdc/bin/1058449141/cockroachdb_cockroach_pkg_sql_tests
mkdir -p benchdiff/2c99066/bin/1058449141
gcloud storage cp gs://cockroach-microbench-ci/builds/2c99066211911e14c7519ec47a5c021488a1fe71/bin/pkg_sql_tests benchdiff/2c99066/bin/1058449141/cockroachdb_cockroach_pkg_sql_tests
chmod +x benchdiff/2c99066/bin/1058449141/cockroachdb_cockroach_pkg_sql_tests
```
**benchdiff command**:
```shell
benchdiff --run=^BenchmarkSysbench/SQL/3node/oltp_read_write$ --old=2c99066 --new=c92abdc ./pkg/sql/tests
```

</details>

</details>
<details><summary><strong>🔴 Sysbench</strong> [KV, 3node, oltp_read_only]</summary>

| Metric                      | Old Commit     | New Commit     | Delta      | Note         |
|-----------------------------|----------------|----------------|------------|--------------|
| 🔴 **sec/op** | 1.057m ±1% | 1.120m ±1% | +5.97% | p=0.000 n=15    |
| 🔴 **allocs/op** | 1.857k ±0% | 2.096k ±0% | +12.87% | p=0.000 n=15    |

<details><summary>Reproduce</summary>

**benchdiff binaries**:
```shell
mkdir -p benchdiff/c92abdc/bin/1058449141
gcloud storage cp gs://cockroach-microbench-ci/builds/c92abdc71850f94fc69cc6e6453605c3896dc280/bin/pkg_sql_tests benchdiff/c92abdc/bin/1058449141/cockroachdb_cockroach_pkg_sql_tests
chmod +x benchdiff/c92abdc/bin/1058449141/cockroachdb_cockroach_pkg_sql_tests
mkdir -p benchdiff/2c99066/bin/1058449141
gcloud storage cp gs://cockroach-microbench-ci/builds/2c99066211911e14c7519ec47a5c021488a1fe71/bin/pkg_sql_tests benchdiff/2c99066/bin/1058449141/cockroachdb_cockroach_pkg_sql_tests
chmod +x benchdiff/2c99066/bin/1058449141/cockroachdb_cockroach_pkg_sql_tests
```
**benchdiff command**:
```shell
benchdiff --run=^BenchmarkSysbench/KV/3node/oltp_read_only$ --old=2c99066 --new=c92abdc ./pkg/sql/tests
```

</details>

</details>
<details><summary><strong>🔴 Sysbench</strong> [KV, 3node, oltp_write_only]</summary>

| Metric                      | Old Commit     | New Commit     | Delta      | Note         |
|-----------------------------|----------------|----------------|------------|--------------|
| 🔴 **sec/op** | 1.911m ±0% | 2.002m ±1% | +4.75% | p=0.000 n=15    |
| 🔴 **allocs/op** | 3.792k ±0% | 4.186k ±0% | +10.39% | p=0.000 n=15    |

<details><summary>Reproduce</summary>

**benchdiff binaries**:
```shell
mkdir -p benchdiff/c92abdc/bin/1058449141
gcloud storage cp gs://cockroach-microbench-ci/builds/c92abdc71850f94fc69cc6e6453605c3896dc280/bin/pkg_sql_tests benchdiff/c92abdc/bin/1058449141/cockroachdb_cockroach_pkg_sql_tests
chmod +x benchdiff/c92abdc/bin/1058449141/cockroachdb_cockroach_pkg_sql_tests
mkdir -p benchdiff/2c99066/bin/1058449141
gcloud storage cp gs://cockroach-microbench-ci/builds/2c99066211911e14c7519ec47a5c021488a1fe71/bin/pkg_sql_tests benchdiff/2c99066/bin/1058449141/cockroachdb_cockroach_pkg_sql_tests
chmod +x benchdiff/2c99066/bin/1058449141/cockroachdb_cockroach_pkg_sql_tests
```
**benchdiff command**:
```shell
benchdiff --run=^BenchmarkSysbench/KV/3node/oltp_write_only$ --old=2c99066 --new=c92abdc ./pkg/sql/tests
```

</details>

</details>

<details><summary>Artifacts</summary>

**download**:
```shell
mkdir -p new
gcloud storage cp gs://cockroach-microbench-ci/artifacts/c92abdc71850f94fc69cc6e6453605c3896dc280/15674654280-1/\* new/
mkdir -p old
gcloud storage cp gs://cockroach-microbench-ci/artifacts/2c99066211911e14c7519ec47a5c021488a1fe71/15674654280-1/\* old/
```

</details>

_built with commit: [c92abdc71850f94fc69cc6e6453605c3896dc280](https://github.com/cockroachdb/cockroach/commit/c92abdc71850f94fc69cc6e6453605c3896dc280)_

<!-- microbench-ci -->

herkolategan commented:
> [cockroach-teamcity](https://github.com/cockroach-teamcity)

Latest post is from the new token.

herkolategan commented:
TFTR!

bors r=rail
